add azure deploy guide (#517)
Quansight/qhub: docs/source/04_how_to_guides/3_qhub-azure-deploy_guide.md

In this section, you need to specify the region and instance types, and number of nodes that you would like to use for deployment. To see available instance types and pricing, please refer to [Azure's documentation](https://azure.microsoft.com/en-us/pricing/details/virtual-machines/linux/).

```yaml
azure:
  project: PLACEHOLDER
  region: Central US
  kubernetes_version: 1.18.14
  node_groups:
    general:
      instance: Standard_D2_v2
      min_nodes: 1
      max_nodes: 1
    user:
      instance: Standard_D2_v2
      min_nodes: 0
      max_nodes: 4
    worker:
      instance: Standard_D2_v2
      min_nodes: 0
      max_nodes: 4
  storage_account_postfix: m5tzjrcg
```

The storage_account_postfix is 8 random, lowercase characters to help ensure that the storage account created by qhub in Azure is unique across all of Azure as required by Azure.

### Default Images

> An image is the term used to refer to a serialized copy of the state of an environment stored in a file. Default images section in the configuration file lists the images used to run the default image if it is not already specified in a *profile* (described in the next section). The `jupyterhub` key controls the jupyterhub image run.

Reviewer: ```suggestion
An **image** is the term used to refer to a serialized copy of the state of an environment stored in a file. The `default images` section in the QHub configuration file lists the images used to run the default image (if not already specified in a *profile* - described in the next section). For example, the `jupyterhub` key controls the JupyterHub image run.
```

```yaml
default_images:
  jupyterhub: "quansight/qhub-jupyterhub:c36eace493739be280c71bec59b80659115db5d5"
  jupyterlab: "quansight/qhub-jupyterlab:c36eace493739be280c71bec59b80659115db5d5"
  dask_worker: "quansight/qhub-dask-worker:c36eace493739be280c71bec59b80659115db5d5"
```

### Storage

> This section allows you to control the amount of storage allocated to shared filesystems. Please note, when you change the storage size, it will delete the previous storage.

Reviewer: ```suggestion
This section controls the amount of storage allocated to shared filesystems.
> NOTE: When the storage size is changed, the previous storage will be deleted.
```

Reviewer: This section also handles the storage for conda-store, right?

```yaml
storage:
  conda_store: 20Gi
  shared_filesystem: 10Gi
```

### Profiles

> Profiles are used to control the jupyterlab user instances and dask-workers provided by dask-gateway.

Reviewer: ```suggestion
Profiles are used to control the `jupyterlab` user instances and `dask-workers` provided by Dask-Gateway.
```
